build/plugins/brand.js

```diff
@@ -39,35 +34,62 @@ const getStyleConfig = () => ({
   altBrandColor: '#5C6BC0'
 });
 
-exports.apply = (api, { files, imagesPath } = {}) => {
-  const brandConfig = loadConfig(api, files) || {};
-  const constants = merge(getAppConfig(), brandConfig);
-  const style = merge(getStyleConfig(), brandConfig.style);
+module.exports = (api, { pluginOptions }) => {
+  const { serve, build } = api.service.commands;
+
+  const serveFn = serve.fn;
+  const buildFn = build.fn;
+
+  serve.fn = (...args) => {
+    addBranding(api, pluginOptions, args[0]);
+    return serveFn(...args);
+  };
+
+  build.fn = (...args) => {
+    addBranding(api, pluginOptions, args[0]);
+    return buildFn(...args);
+  };
+};
+
+function addBranding(api, { brand }, { brandConfig }) {
+  const { files, imagesPath } = brand;
+  console.log(brandConfig);
+  const config = loadConfig(brandConfig, files) || {};
+  const constants = merge(getAppConfig(), config);
+  const style = merge(getStyleConfig(), config.style);
+
+  const { projectOptions } = api.service;
 
-  const { config } = api;
-  config.output.html = Object.assign({}, config.output.html, {
+  projectOptions.pages.index = Object.assign({}, projectOptions.pages.index, {
     title: constants.title,
     favicon: path.join('client/', imagesPath, constants.favicon)
   });
-  config.css = Object.assign({}, config.css, {
-    loaderOptions: {
-      sass: { data: map(style, toScssVariable).join('\n') }
-    }
+
+  projectOptions.css.loaderOptions = Object.assign({}, projectOptions.css.loaderOptions, {
+    sass: { data: map(style, toScssVariable).join('\n') }
   });
-  config.constants = Object.assign({}, config.constants, {
-    BRAND_CONFIG: JSON.stringify({
-      TITLE: constants.title,
-      FAVICON: path.join(imagesPath, constants.favicon),
-      LOGO_COMPACT: path.join(imagesPath, constants.logo.compact),
-      LOGO_FULL: path.join(imagesPath, constants.logo.full)
-    })
+
+  api.service.projectOptions = projectOptions;
+  console.log(projectOptions.pages.index);
+  console.log(constants);
+
+  api.chainWebpack(webpackConfig => {
```

Comment:
Don't be _over_-descriptive. Both `vue-cli` docs and `webpack-chain` docs call it simply `config` exclusively. I understand that this is more descriptive but it also makes people familiar with convention stop and think is that really a chainable config or something else.